### PR TITLE
3 commits, mostly about lints

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -10,29 +11,69 @@ linter:
   rules:
     - annotate_overrides
     - avoid_empty_else
+    - avoid_function_literals_in_foreach_calls
```
Comment (pq) [inline analysis_options.yaml]: This is a lot of lints.  Is this a group you've used elsewhere?  If it's shared, maybe we should consider putting the set someplace shareable the way package `pedantic` does?

File: lib/logging.dart [hunk 2]
```diff
@@ -52,7 +52,7 @@ class Logger {
   /// Singleton constructor. Calling `new Logger(name)` will return the same
   /// actual instance whenever it is called with the same string name.
   factory Logger(String name) {
-    return _loggers.putIfAbsent(name, () => new Logger._named(name));
+    return _loggers.putIfAbsent(name, () => Logger._named(name));
```
Comment (pq) [inline lib/logging.dart]: While you're here, perhaps the short form/ . (`=>`?)

File: lib/logging.dart [hunk 3]
```diff
@@ -226,9 +226,7 @@ class Logger {
 
   Stream<LogRecord> _getStream() {
     if (hierarchicalLoggingEnabled || parent == null) {
-      if (_controller == null) {
-        _controller = new StreamController<LogRecord>.broadcast(sync: true);
-      }
+      _controller ??= StreamController<LogRecord>.broadcast(sync: true);
```
Comment (pq) [inline lib/logging.dart]: 👍

File: test/logging_test.dart [hunk 4]
```diff
@@ -175,8 +175,8 @@ void main() {
     test('child zone', () {
       var root = Logger.root;
 
-      var recordingZone;
-      var records = new List<LogRecord>();
+      Zone recordingZone;
```
Comment (pq) [inline test/logging_test.dart]: I admit I'm confused.  Types are elided in some places and then added here...

Is this one right?

File: .travis.yml [hunk 5]
```diff
@@ -6,18 +6,13 @@ dart:
...
   - test: --platform vm
-  - test: --platform firefox
   - test: --platform chrome
-  - dartanalyzer
+  - dartanalyzer: --fatal-infos --fatal-warnings .
 
 matrix:
   include:
     - dart: dev
       dart_task: dartfmt
```
[the remark this answers is not in the document]
Comment (kevmoo) [reply .travis.yml]: I don't see any failures here. Am I missing something?

File: analysis_options.yaml (hunk 1 above)
Comment (kevmoo) [reply analysis_options.yaml]: Yeah...this is a group of things we've been pushing around through a lot of packages I work with – build and friends.

File: lib/logging.dart (hunk 2 above)
Comment (kevmoo) [reply lib/logging.dart]: Trying to stick w/ just lints here – to make review easier.

Happy to do that in a follow-up

File: test/logging_test.dart (hunk 4 above)
Comment (kevmoo) [reply test/logging_test.dart]: Yup. `var` when inference work, be explicit otherwise. We cannot infer the type `Zone` here.

**Comment by kevmoo** [comment]
@pq – LGTM, or no?

File: .travis.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (pq) [reply .travis.yml]: Ah, sorry.  Not a failure quite but what's up with this spam?

    dartfmt only supports 'sdk' as an optional argument value.

https://travis-ci.org/dart-lang/logging/jobs/447188192#L505

File: analysis_options.yaml (hunk 1 above)
Comment (pq) [reply analysis_options.yaml]: Have you considered capturing them in one place?  How do you maintain them?  Is there any discussion we could link to somewhere?

File: analysis_options.yaml [hunk 6]
```diff
@@ -10,29 +11,69 @@ linter:
...
+    - avoid_relative_lib_imports
     - avoid_return_types_on_setters
+    - avoid_returning_null
+    - avoid_types_as_parameter_names
+    - avoid_unused_constructor_parameters
     - await_only_futures
     - camel_case_types
+    - cancel_subscriptions
     - comment_references
+    #- constant_identifier_names
```
Comment (pq) [inline analysis_options.yaml]: Perhaps add a link to https://github.com/dart-lang/logging/issues/43?

File: test/logging_test.dart (hunk 4 above)
Comment (pq) [reply test/logging_test.dart]: Ugh, yeah.  That's what I thought.

File: pubspec.yaml [hunk 7]
```diff
@@ -1,15 +1,16 @@
...
+description: >-
   Provides APIs for debugging and error logging. This library introduces
   abstractions similar to those used in other languages, such as the Closure
   JS Logger and java.util.logging.Logger.
 author: Dart Team <misc@dartlang.org>
 homepage: https://github.com/dart-lang/logging
 
 environment:
-  sdk: '>=1.5.0 <3.0.0'
+  sdk: '>=2.0.0 <3.0.0'
```
Comment (pq) [inline pubspec.yaml]: Does this merit a `CHANGELOG` update?

File: pubspec.yaml [hunk 8]
```diff
@@ -1,15 +1,16 @@
...
 
 dev_dependencies:
-  test: '>=0.12.0 <2.0.0'
+  pedantic: ^1.3.0
```
Comment (pq) [inline pubspec.yaml]: It seems that in general folks are pinning versions when pulling in `pedantic`.  For example:

https://github.com/dart-lang/test/blob/957a0f8fbd3dc21de746d448ff313c148b3e51c7/pkgs/test/pubspec.yaml#L48

Are you doing differently elsewhere?

File: analysis_options.yaml [hunk 9]
```diff
@@ -10,29 +11,69 @@ linter:
   rules:
     - annotate_overrides
     - avoid_empty_else
```
Comment (pq) [inline analysis_options.yaml]: This is in `pedantic` and can be removed.

File: analysis_options.yaml [hunk 10]
```diff
@@ -10,29 +11,69 @@ linter:
...
+    - avoid_function_literals_in_foreach_calls
     - avoid_init_to_null
+    - avoid_null_checks_in_equality_operators
+    - avoid_relative_lib_imports
```
Comment (pq) [inline analysis_options.yaml]: This is in pedantic and can be removed.

File: analysis_options.yaml [hunk 11]
```diff
@@ -10,29 +11,69 @@ linter:
...
     - avoid_return_types_on_setters
```
Comment (pq) [inline analysis_options.yaml]: This is in pedantic and can be removed.

File: analysis_options.yaml [hunk 12]
```diff
@@ -10,29 +11,69 @@ linter:
...
     - empty_statements
     - hash_and_equals
     - implementation_imports
+    - invariant_booleans
+    - iterable_contains_unrelated_type
     - library_names
     - library_prefixes
+    - list_remove_unrelated_type
+    - no_adjacent_strings_in_list
+    - no_duplicate_case_values
```
Comment (pq) [inline analysis_options.yaml]: Could you remove all the `pedantic` dups?

```
    - no_duplicate_case_values
    - prefer_contains
    - prefer_equal_for_default_values
    - prefer_is_not_empty
    - recursive_getters
    - unrelated_type_equality_checks
    - use_rethrow_when_possible
    - unawaited_futures
    - valid_regexps
```

See: https://github.com/dart-lang/pedantic/blob/v1.3.0/lib/analysis_options.yaml

File: analysis_options.yaml (hunk 1 above)
Comment (kevmoo) [reply analysis_options.yaml]: Not really...although it wouldn't be a bad idea.

File: analysis_options.yaml (hunk 6 above)
Comment (kevmoo) [reply analysis_options.yaml]: ack

File: pubspec.yaml (hunk 7 above)
Comment (kevmoo) [reply pubspec.yaml]: ack

File: pubspec.yaml (hunk 8 above)
Comment (kevmoo) [reply pubspec.yaml]: Our thought is that we should not be pinning – so we pick up the latest as things evolve. If that means travis breaks and we have to fix – it's okay.

File: .travis.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (kevmoo) [reply .travis.yml]: It's some weirdness w/ the travis rule – no affect on behavior.

**Comment by kevmoo** [comment]
@pq – PTAL

**Pull request merged**